### ISSUE TITLE
kern error: (os/kern) failure

### ISSUE THREAD

**Issue by 1337kingOf** [notahacker8/MacInject#1]
on mac m1
trying to inject to any process
this dylib
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <CoreFoundation/CoreFoundation.h>
#include <CoreGraphics/CoreGraphics.h>
#include <CoreServices/CoreServices.h>
#include <ApplicationServices/ApplicationServices.h>

// Function to display a pop-up window
void displayPopup() {
    // Create a CFStringRef containing your message
    CFStringRef message = CFStringCreateWithCString(NULL, "All ok", kCFStringEncodingUTF8);

    // Create a dictionary for the user notification
    CFMutableDictionaryRef dict = CFDictionaryCreateMutable(NULL, 0, &kCFTypeDictionaryKeyCallBacks, &kCFTypeDictionaryValueCallBacks);
    CFDictionaryAddValue(dict, kCFUserNotificationAlertMessageKey, message);

    // Create a user notification
    SInt32 error;
    CFUserNotificationRef userNotification = CFUserNotificationCreate(NULL, 0, kCFUserNotificationPlainAlertLevel, &error, dict);

    // Display the notification
    CFOptionFlags response;
    CFUserNotificationReceiveResponse(userNotification, 0, &response);

    // Clean up
    CFRelease(message);
    CFRelease(dict);
    CFRelease(userNotification);
}

// Entry point when the library is loaded
__attribute__((constructor))
void libraryLoad() {
    displayPopup();
}

and getting error

**Comment by notahacker8** [comment]
Hi, is the target process running natively or running under Rosetta?

**Comment by notahacker8** [comment]
I updated the code to show what line the error is on.

**Comment by 1337kingOf** [comment]
how to understand , rosetta or no?

**Comment by 1337kingOf** [comment]
and by the way do u know how to dump local items keychain, or decrypt it with no every time ask for every item password

**Comment by notahacker8** [comment]
> how to understand , rosetta or no?

The injector for arm64 doesn't work on intel applications running on apple silicon.

> and by the way do u know how to dump local items keychain, or decrypt it with no every time ask for every item password

No, but for my project, I added the debugging tool entitlement in Xcode to allow for injection without entering a password every time.

Also, the injector will not work if you run in Xcode. You will have to build it first, then use it.

**Comment by 1337kingOf** [comment]
how to check process arm or no ?

**Comment by 1337kingOf** [comment]
yeah im injecting my arm app, arm library to arm arch, and not working

**Comment by notahacker8** [comment]
> how to check process arm or no ?

Activity monitor will show if an app is Intel or Apple

The most recent version of the code should show you what line the error is on now.

**Comment by 1337kingOf** [comment]
give ur telegram
